clang/runtime/dpct-rt/include/dpct/dpl_extras/dpcpp_extensions.h

```diff
@@ -566,6 +566,89 @@ class radix_sort {
   uint8_t *_local_memory;
 };
 
+/// Load linear segment items into block format across threads
+/// Helper for Block Load
+enum load_algorithm {
+
+  BLOCK_LOAD_DIRECT,
+  BLOCK_LOAD_STRIPED,
+  // To-do: BLOCK_LOAD_WARP_TRANSPOSE
+
+};
+
+// loads a linear segment of workgroup items into a subgroup striped
+// arrangement. Created as free function until exchange mechanism is
+// implemented.
+// To-do: inline this function with BLOCK_LOAD_WARP_TRANSPOSE mechanism
+template <size_t GROUP_WORK_ITEMS, size_t ITEMS_PER_WORK_ITEM, typename InputT,
+          typename InputIteratorT, typename Item>
+__dpct_inline__ void
+load_subgroup_striped(const Item &item, size_t linear_tid,
+                      InputIteratorT block_itr,
+                      InputT (&items)[ITEMS_PER_WORK_ITEM]) {
+
+  // This implementation does not take in account range loading across
+  // workgroup items To-do: Decide whether range loading is required for group
+  // loading
+
+  size_t subgroup_offset = item.get_sub_group().get_local_range()[0];
+#pragma unroll
+  for (uint32_t idx = 0; idx < ITEMS_PER_WORK_ITEM; idx++) {
+    new (&items[idx]) InputT(
```

Comment:
I'm not sure if this was answered before, but why does this function differ in semantics to create items with placement `new`, where the others just assign?
Is this the desired effect?  Will load_subgroup_striped always be loading into uninitialized memory where the others are not?
If so, do we need to adjust naming / comments to make this clear?